VetClinic.API/Controllers/ClientController.cs

```diff
@@ -0,0 +1,89 @@
+﻿using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using VetClinic.DAL.Entities;
+using VetClinic.API.DTO.ClientDto;
+using VetClinic.BLL.Services.Interfaces;
+using VetClinic.API.DTO.ClientDto;
+
+namespace VetClinic.API.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class ClientController : ControllerBase
+    {
+        private IClientService _clientService;
+        private IMapper _mapper;
+     
+        public ClientController(IClientService clientService, IMapper mapper)
+        {
+            _clientService = clientService;           
+            _mapper = mapper;
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> PostAsync(CreateClientDto dto)
+        {
+            User user =  _mapper.Map<CreateClientDto, User>(dto);
+            Client client = new Client();
```

Comment:
Why do you create an empty client in the controller?
It may be better to take it to the service...